src/main/java/uk/gov/hmcts/ccd/data/casedetails/search/SortOrderField.java

```diff
@@ -0,0 +1,16 @@
+package uk.gov.hmcts.ccd.data.casedetails.search;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+
+@Getter
+@AllArgsConstructor
+@Builder(builderMethodName = "sortOrderWith")
+public class SortOrderField {
+
+    private String caseFieldId;
+    private boolean metadata;
```

Comment:
SortOrderField type is added to the Metadata class. So in theory we would expect these to be always metadata. 
With this solution we are adding case fields info which is not metadata to the Metadata class. Can be confusing, but I appreciate it simplifies things a bit.
If we wanted to be consistent perhaps we should have added case fields to the queryParameters?

Reply:
well I thought about it but current one actually simplifies with minimal changes. 
1. if we pass some sortFields to metaData and some to queryParameters, we've to consolidate again with proper ordering in the repository. 
2. atm queryParameters is just a Map<String, String>, so we've to change signature to Map<String, Object> or something else 

Another approach change signature of SearchOperation.execute(MetaData metaData, Map<String, String> criteria) to
` execute(MetaData metaData, Map<String, String> criteria, SortOrderField... sortOrderField))` or best wrap it to a new object `execute(SearchCriteria, searchCriteria)`. 
But this one requires significant refactoring 🤔